doc/akri_connector/connector-metadata-schema.json

```diff
@@ -0,0 +1,328 @@
+          ]
+        }
+      ]
+    },
+    "kubernetes": {
+      "type": "object",
+      "description": "The kubernetes-level metadata",
+      "properties": {
+        "minimumKubernetesVersion": {
+          "type": "string",
+          "description": "Minimum required version of kubernetes needed to run this connector"
+        }
+      }
+    },
+    "secrets": {
+      "type": "array",
+      "description": "The array of secrets that are required for the connector application to work. The secret will be mounted onto all connector instances deployed for this connector application",
+      "items": [
+        {
+          "type": "object",
+          "description": "A single secret that will be mounted onto all connector instances.",
+          "properties": {
+            "secretAlias": {
+              "type": "string",
+              "description": "The application-defined alias for the secret"
+            },
+            "secretDescription": {
+              "type": "string",
+              "description": "The description for this secret alias"
+            },
+            "isOptional": {
+              "type": "boolean",
+              "description": "If the secret is optional or mandatory",
+              "default": false
+            }
+          },
+          "required": [
+            "secretAlias"
+          ]
+        }
+      ]
+    }
+  },
+  "supportedArchitectures": {
+    "type": "array",
+    "description": "The CPU architectures that this connector image was built for. At least one must be specified.",
+    "items": [
+      {
+        "type": "string",
+        "description": "One of the CPU architectures this connector image was built for. For example: 'linux/amd64', 'linux/arm64', or 'windows/amd64'"
+      }
+    ],
+    "minItems": 1,
+    "uniqueItems": true
+  },
+  "sourceCode": {
+    "type": "object",
+    "description": "Details about the source code that compiled this connector",
+    "properties": {
+      "language": {
+        "type": "string",
+        "description": "The programming language used to write this connector"
+      },
+      "languageVersion": {
+        "type": "string",
+        "description": "The version of the programming language used to write this connector"
+      },
+      "sdks": {
+        "type": "object",
+        "description": "Details about the versions of the AIO SDKs used to write this connector (if they were used)",
+        "properties": {
+          "mqttPackageVersion": {
+            "type": "string",
+            "description": "The version of the Azure IoT Operations MQTT package"
+          },
+          "servicesPackageVersion": {
+            "type": "string",
+            "description": "The version of the Azure IoT Operations services package"
+          },
+          "connectorPackageVersion": {
+            "type": "string",
+            "description": "The version of the Azure IoT Operations connector package"
+          }
+        }
+      }
```

Comment:
what is the purpose of this field?

we use also a azure_iot_operations_protocol SDK.

to be honest, if there is no good reason we should not keep this. do we need the sourceCode field at all?

Reply:
Added the protocol field as well. You brought up this same question above. It's moreso for us who write the connectors so that we can easily debug any issue logged against the connector image.